static/app/components/group/tagDistributionMeter.tsx

```diff
@@ -15,67 +14,24 @@ type Props = {
   totalValues: number;
 };
 
-type State = {
-  error: boolean;
-  loading: boolean;
-  iOSDeviceList?: IOSDeviceList;
-};
-
-class GroupTagDistributionMeter extends Component<Props, State> {
-  state: State = {
-    loading: true,
-    error: false,
-  };
-
-  componentDidMount() {
-    this.fetchData();
-  }
-
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
+class GroupTagDistributionMeter extends Component<Props> {
+  shouldComponentUpdate(nextProps: Props) {
     return (
-      this.state.loading !== nextState.loading ||
-      this.state.error !== nextState.error ||
       this.props.tag !== nextProps.tag ||
       this.props.name !== nextProps.name ||
       this.props.totalValues !== nextProps.totalValues ||
       this.props.topValues !== nextProps.topValues
     );
   }
 
-  fetchData() {
-    this.setState({
-      loading: true,
-      error: false,
-    });
-
-    loadDeviceListModule('iOS')
-      .then(iOSDeviceList => {
-        this.setState({
-          iOSDeviceList,
-          error: false,
-          loading: false,
-        });
```

Comment:
The act was likely happening to due to this `setState`... we can just have a mounted ref and check it before doing `setState` and I'm guessing the act would go away.